src/Google.Storage.V1/StorageClient.ListBuckets.cs

```diff
@@ -65,18 +55,9 @@ public partial class StorageClient
         /// <param name="options">The options for the operation. May be null, in which case
         /// defaults will be supplied.</param>
         /// <returns>A sequence of buckets within the project.</returns>
-        public IEnumerable<Bucket> ListBuckets(string projectId, ListBucketsOptions options = null)
-        {
-            Preconditions.CheckNotNull(projectId, nameof(projectId));
-            var initialRequest = CreateListBucketsRequest(projectId, options);
-            return s_bucketPageStreamer.Fetch(initialRequest);
-        }
-
-        private BucketsResource.ListRequest CreateListBucketsRequest(string projectId, ListBucketsOptions options)
+        public virtual IEnumerable<Bucket> ListBuckets(string projectId, ListBucketsOptions options = null)
```

Comment:
Because the sync version is lazy and partial - you can stop fetching at any time.

ListAllBucketsAsync is asynchronous, but fetches everything - because the page streaming support in GoogleApis.Core doesn't include IAsyncEnumerable.